### ISSUE TITLE
Meta key definition

### ISSUE THREAD

**Issue by Arifi** [twoelevenjay/CMB2-Admin-Extension#41]
in one of my projects i used CMB2-Admin-Ext to create a gallery filed. then i wanted to extract those images in a separate single.php file, so i used this code :
`$images = get_post_meta( $post->ID, '_photos', true);`
the meta key `_photos ` was foun in the database, it will be good if you add a field in CMB2-Admin-Ext to allow the admin define the meta key because actually i'm spending a lot of time searching how this plugin defines the meta key

**Comment by twoelevenjay** [comment]
You are right, I have not made this clear. I will add it in the settings and make some documentation.

It takes the field name, converts it to snake case, and prepends it with an underscore and uses that as the key:

Leon Shelhamer > _leon_shelhamer

You can also inspect the field element in the browser to see what is being used for the "name" prop.

![cmbr-admin-ext](https://user-images.githubusercontent.com/4624787/32497741-91a2e3a4-c39b-11e7-85fa-2403c4fcf8d6.png)

**Comment by twoelevenjay** [comment]
As far as adding a field to control the key, I am worried this will cause more trouble than it will help. If a key gets changed after fields have been added to the database then the fields will all be lost. Unless the plugin is written to do a search and replace on the DB anytime the "meta key" field is changed. Do you really think this is needed? Or would better documentation on how to use the field be sufficient?

**Comment by Arifi** [comment]
Hi,
Sorry i didnt responded earlier.. this is verry nice, at least now we know the key
See here how ACF handles fields, you can maybe allow users to define the key name ?

![image](https://user-images.githubusercontent.com/7891131/34323447-ea2aac6e-e849-11e7-8a30-06b436ac6947.png)
![image](https://user-images.githubusercontent.com/7891131/34323452-0f5f8aa4-e84a-11e7-9d52-130605a9a0b5.png)
